src/pipeline_configuration_spec.py

```diff
@@ -30,45 +27,149 @@
+                           Stages that need to sync participant urns back to another source will use this table
+                           to re-identify participants as late as possible.
+                           All other stages will only use participant uuids, and will not even load the uuid table.
+        :type uuid_table: src.common.configuration.UUIDTableClientConfiguration
+        :param operations_dashboard: Configuration for the OperationsDashboard. Pipeline start and end events will
+                                     be logged to this dashboard.
+        :type operations_dashboard: src.common.configuration.OperationsDashboardConfiguration
+        :param archive_configuration: Configuration for uploading the pipeline run archives.
+                                      Under normal operation, analysis outputs, traced data, and caches will be uploaded
+                                      to a Google Cloud Storage bucket once per day.
+        :type archive_configuration: src.common.configuration.ArchiveConfiguration
+        :param description: Optional natural language description of what this configuration is for.
+                            This field is not used by any pipeline stage, it exists solely to encourage good
+                            documentation of pipeline configuration files.
+        :type description: str | None
+        :param project_start_date: Messages sent before this date will not be synced from Rapid Pro to the engagement
+                                   database.
+                                   TODO: Only the Rapid Pro -> DB stage uses this. Either move this configuration to
+                                         that stage, or update the other stages to use this too.
+        :type project_start_date: datetime.datetime | None
+        :param project_end_date: Messages sent on or after this date will not be synced from Rapid Pro to the
+                                 engagement database.
+                                 TODO: Only the Rapid Pro -> DB stage uses this. Either move this configuration to
+                                         that stage, or update the other stages to use this too.
+        :type project_end_date: datetime.datetime | None
+        :param test_participant_uuids: List of participant uuids that identify test users.
+                                       These test users will be excluded from analysis.
+                                       TODO: Only the DB -> analysis stage uses this. Either move this configuration to
+                                             that stage, or update the other stages to use this too.
+        :type test_participant_uuids: (list of str) | None
+        :param rapid_pro_sources: Configuration for Rapid Pro -> engagement database syncs.
+        :type rapid_pro_sources: (list of RapidProSource) | None
+        :param facebook_sources: Configuration for Facebook -> engagement database syncs.
```

Comment:
I don't think we should add detail on a case-by-case basis here because it's easy to look up the documentation for each integration for more details, and if we duplicate that here it might get out of sync with future changes we make to the integrations themselves.

If you think we need a high-level overview of how the integrations work in general then I'd be open to adding that - what do you think?